<?xml version="1.0" encoding="UTF-8" ?>
<launch>
    
    <arg name="map" default="vrep3D"/>
    <arg name="simulation" default="true"/>
    <arg name="robot_name1" default="ugv1" />
    <arg name="robot_name2" default="ugv2" />
    <arg name="simulator" default="vrep" />
    <arg name="laser_mapper_running" default="false" />
    <arg name="respawn_value" default="false" /> <!-- boolean: true, false -->

  <node name="robot_trajectory_saver_node" pkg="robot_trajectory_saver" type="robot_trajectory_saver_node" output="screen">
      
    <param name="map" value="$(find patrolling3d_sim)/maps/$(arg map)/$(arg map).dot"/>
    <param name="run_in_simulation" value="$(arg simulation)"/>
    <param name="run_laser_mapper" value="$(arg laser_mapper_running)"/>
    
    
<!--
    <param name="run_in_simulation" value="true"/>
    <param name="num_robots" value="2"/>
    <param name="odom_frame" value="odom"/>
    <param name="robot_name_prefix" value="ugv"/>
    <param name="robot_frame" vale="base_link"/>
    <param name="global_frame" value="/map"/>
    <param name="imu_odom_topic" value="odom"/>

    <param name="frequency" value="1"/>
    
    <param name="leaf_size" value="1"/>
    
    <param name="min_radius" value="1"/>
    <param name="max_radius" value="3"/>
    <param name="radius_incr" value="0.25"/>
    <param name="scaled_factor" value="4./3."/>
    
    <param name="trajectories_marker_topic" value="robot_trajectories_marker"/>
    <param name="save_robot_trajectories_service_name" value="save_on_file_robot_trajectories_only"/>
    <param name="get_robot_trajectories_service_name" value="get_robot_trajectories_nav_msgs"/>
    <param name="load_robot_trajectories_service_name" value="load_robot_trajectories_only"/>


-->
  </node>

<!-- fake service call -->
<!-- <node name="test_build_graph_services1" pkg="robot_trajectory_saver" type="test_build_graph_services" output="screen"/> -->
<!--  <node name="test_build_graph_services2" pkg="robot_trajectory_saver" type="test_build_graph_services" output="screen"/> -->


    <!-- normals computation -->
    <!--
    <node name="compute_normals_$(arg robot_name1)" pkg="path_planner" type="compute_normals" respawn="$(arg respawn_value)" output="screen">
	<rosparam file="$(find  path_planner)/launch/path_planner_octomap.yaml" /> 
		
        <param name = "robot_frame_name" value = "/$(arg robot_name1)/base_link"/>
        <param name = "laser_frame_name" value = "/$(arg robot_name1)/laser"/>
					
	 
        <remap from = "cloud_in" to = "/volumetric_mapping/octomap_pcl"/>
        
	<remap from = "cloud_out" to = "/$(arg simulator)/$(arg robot_name1)/local_map_normals"/>
	<remap from = "/normals_marker" to = "/$(arg simulator)/$(arg robot_name1)/normal_markers"/>
		
    </node> 
    -->
</launch>
